nuclio_sdk/event.py

```diff
@@ -130,29 +126,102 @@ def from_parsed_data(cls, parsed_data, body, content_type):
             version=parsed_data["version"],
         )
 
+    @classmethod
+    def from_parsed_data_bytes(cls, parsed_data, body):
+        trigger = TriggerInfo(
+            parsed_data[b"trigger"][b"kind"], parsed_data[b"trigger"][b"name"]
+        )
+        return cls(
+            body=body,
+            content_type=parsed_data[b"content_type"],
+            trigger=trigger,
+            fields=parsed_data[b"fields"],
+            headers=parsed_data[b"headers"],
+            _id=parsed_data[b"id"],
+            method=parsed_data[b"method"],
+            path=parsed_data[b"path"],
+            size=parsed_data[b"size"],
+            timestamp=datetime.datetime.utcfromtimestamp(parsed_data[b"timestamp"]),
+            url=parsed_data[b"url"],
+            shard_id=parsed_data[b"shard_id"],
+            num_shards=parsed_data[b"num_shards"],
+            _type=parsed_data[b"type"],
+            type_version=parsed_data[b"type_version"],
+            version=parsed_data[b"version"],
+        )
+
+    def __repr__(self):
+        return self.to_json()
+
+
+class EventDeserializer(object):
+    def deserialize(self, event_message) -> Event:
```

Comment:
Maybe it's best to wait for decisions wrt type hints?
Regardless I have slight beef with the deserializer and its factory being exposed to the user. it can all be in internal helpers, the user don't need anything more than `Event.deserialize(<input>, which-type)`

Reply:
Decided to use enum values